### PR TITLE
fix vs2010 build error

### LINKED ISSUE #754: errors on build in vs2010
Fix: #755
Version: 3.1.0-1195-gbf4c5be
Errors:
"C:\Lib\opencv\build_x86_pre\OpenCV.sln" (default target) (1) ->
"C:\Lib\opencv\build_x86_pre\ALL_BUILD.vcxproj.metaproj" (default target) (2) -
>
"C:\Lib\opencv\build_x86_pre\modules\optflow\opencv_optflow.vcxproj.metaproj" (
default target) (64) ->
"C:\Lib\opencv\build_x86_pre\modules\optflow\opencv_optflow.vcxproj" (default t
arget) (67) ->
  ..\..\..\sources\opencv_contrib\modules\optflow\src\dis_flow.cpp(1069): error
 C2668: 'cv::pow' : ambiguous call to overloaded function [C:\Lib\opencv\build_
x86_pre\modules\optflow\opencv_optflow.vcxproj]
  ..\..\..\sources\opencv_contrib\modules\optflow\src\pcaflow.cpp(229): error C
2668: 'cv::sqrt' : ambiguous call to overloaded function [C:\Lib\opencv\build_x
86_pre\modules\optflow\opencv_optflow.vcxproj]
  ..\..\..\sources\opencv_contrib\modules\optflow\src\pcaflow.cpp(299): error C
2668: 'cv::sqrt' : ambiguous call to overloaded function [C:\Lib\opencv\build_x
86_pre\modules\optflow\opencv_optflow.vcxproj]

"C:\Lib\opencv\build_x86_pre\OpenCV.sln" (default target) (1) ->
"C:\Lib\opencv\build_x86_pre\ALL_BUILD.vcxproj.metaproj" (default target) (2) -
>
"C:\Lib\opencv\build_x86_pre\modules\xphoto\opencv_xphoto.vcxproj.metaproj" (de
fault target) (106) ->
"C:\Lib\opencv\build_x86_pre\modules\xphoto\opencv_xphoto.vcxproj" (default tar
get) (107) ->
  ..\..\..\sources\opencv_contrib\modules\xphoto\src\learning_based_color_balan
ce.cpp(464): error C2668: 'cv::log' : ambiguous call to overloaded function [C:
\Lib\opencv\build_x86_pre\modules\xphoto\opencv_xphoto.vcxproj]
  ..\..\..\sources\opencv_contrib\modules\xphoto\src\learning_based_color_balan
ce.cpp(464): error C2668: 'cv::log' : ambiguous call to overloaded function [C:
\Lib\opencv\build_x86_pre\modules\xphoto\opencv_xphoto.vcxproj]
  ..\..\..\sources\opencv_contrib\modules\xphoto\src\learning_based_color_balan
ce.cpp(464): error C3861: 'round': identifier not found [C:\Lib\opencv\build_x8
6_pre\modules\xphoto\opencv_xphoto.vcxproj]

### PR DESCRIPTION
**Pull request by Bleach665**
resolves #754

### CODE REVIEW

File: modules/xphoto/src/learning_based_color_balance.cpp [hunk 1]
```diff
@@ -461,7 +461,8 @@ void extractSimpleFeatures(InputArray _src, OutputArray _dst, int range_max_val,
 inline float regressionTreePredict(Vec2f src, uchar *tree_feature_idx, float *tree_thresh_vals, float *tree_leaf_vals)
 {
     int node_idx = 0;
-    int depth = (int)round(log(num_tree_nodes) / log(2));
+	//result always positive, so we can replace round as static_cast + .5
+    int depth = static_cast<int>( (log(static_cast<float>(num_tree_nodes)) / log(static_cast<float>(2))) + .5);
```
Comment (mshabunin) [inline modules/xphoto/src/learning_based_color_balance.cpp]: Can we use _cvRound_ here?

**Comment by mshabunin** [comment]
@sbokov , please take a look.

File: modules/xphoto/src/learning_based_color_balance.cpp (hunk 1 above)
Comment (sbokov) [inline modules/xphoto/src/learning_based_color_balance.cpp]: Sure, cvRound should work as well. I would also replace "static_cast<float>(2)" with simply "2.0f".

**Comment by Bleach665** [comment]
commit updated

**Comment by mshabunin** [comment]
@Bleach665 , please rebase your changes to resolve merging conflicts.

**Comment by Bleach665** [comment]
@mshabunin, I do not understand what I should do(

**Comment by mshabunin** [comment]
Try following commands:
```
# assuming upstream = github.com/opencv, origin = github.com/Bleach665
git fetch upstream master
git rebase upstream/master
git push -f origin master
```

**Comment by StevenPuttemans** [comment]
Just want to add a notice that building the recent masters of both repos on Ubuntu works just fine, so I am wondering why code like this only generates errors on Windows and not on Ubuntu?

**Comment by StevenPuttemans** [comment]
That could indeed be it!

**Comment by alalek** [comment]
Buildbot doesn't show exact reason of conflict. It shows just file name.

In file learning_based_color_balance.cpp "depth" was removed and was replaced with "tree_depth":
```
<<<<<<< 90fe437ecffa312cfb402f405442b6ce587f4bd1
    for (int i = 0; i < tree_depth; i++)
||||||| merged common ancestors
    int depth = (int)round(log(num_tree_nodes) / log(2));
    for (int i = 0; i < depth; i++)
=======
    int depth = cvRound( (log(static_cast<float>(num_tree_nodes)) / log(2.0f)) );
    for (int i = 0; i < depth; i++)
>>>>>>> fix vs2010 build error
```

Run these commands to update your patch:
```
git pull -v --rebase  https://github.com/opencv/opencv_contrib master
# edit file
git add modules/xphoto/src/learning_based_color_balance.cpp
git rebase --continue
git push -f origin master
```

**Comment by mshabunin** [comment]
@Bleach665 , I've [rebased your fix](https://github.com/opencv/opencv_contrib/compare/master...mshabunin:pr756). Please take a look and update your pull request.

**Comment by Bleach665** [comment]
@alalek, @mshabunin. Thanks you for the clarification.

**Comment by mshabunin** [comment]
Moved to #763

**Pull request closed without merging**